java/CMakeLists.txt

```diff
@@ -112,18 +112,123 @@ set(NATIVE_JAVA_CLASSES
         org.rocksdb.WriteOptions
 )
 
-include_directories($ENV{JAVA_HOME}/include)
-include_directories($ENV{JAVA_HOME}/include/win32)
-include_directories(${PROJECT_SOURCE_DIR}/java)
-
 set(JAVA_TEST_LIBDIR ${PROJECT_SOURCE_DIR}/java/test-libs)
 set(JAVA_TMP_JAR ${JAVA_TEST_LIBDIR}/tmp.jar)
 set(JAVA_JUNIT_JAR ${JAVA_TEST_LIBDIR}/junit-4.12.jar)
 set(JAVA_HAMCR_JAR ${JAVA_TEST_LIBDIR}/hamcrest-core-1.3.jar)
 set(JAVA_MOCKITO_JAR ${JAVA_TEST_LIBDIR}/mockito-all-1.10.19.jar)
 set(JAVA_CGLIB_JAR ${JAVA_TEST_LIBDIR}/cglib-2.2.2.jar)
 set(JAVA_ASSERTJ_JAR ${JAVA_TEST_LIBDIR}/assertj-core-1.7.1.jar)
-set(JAVA_TESTCLASSPATH "${JAVA_JUNIT_JAR}\;${JAVA_HAMCR_JAR}\;${JAVA_MOCKITO_JAR}\;${JAVA_CGLIB_JAR}\;${JAVA_ASSERTJ_JAR}")
+set(JAVA_TESTCLASSPATH ${JAVA_JUNIT_JAR} ${JAVA_HAMCR_JAR} ${JAVA_MOCKITO_JAR} ${JAVA_CGLIB_JAR} ${JAVA_ASSERTJ_JAR})
+
+include(UseJava)
+include(FindJNI)
+
+add_jar(
+  rocksdbjni_classes
+  SOURCES
+  src/main/java/org/rocksdb/AbstractCompactionFilter.java
```

Comment:
Uuugh! That seems really like a shortcoming in CMake. It would seem to me that it should do a quick check of the timestamp of the source files and determine if any are newer than the last time CMake was run.

I think previously we were using the GLOB right? Do you guys need this regenerate facility?